test/unit/utils/jwt-utils.test.ts

```diff
@@ -0,0 +1,108 @@
+import jwt from 'jsonwebtoken';
+import { isJWTExpired } from '../../../app/utils/jwt-utils';
+import { expect, sinon } from '../../utils/testUtils';
+
+describe('JWT Utils', () => {
+  let sandbox: sinon.SinonSandbox;
+
+  beforeEach(() => {
+    sandbox = sinon.createSandbox();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('checks that a token is expired', () => {
+
+    let dateOffset = (24 * 60 * 60 * 1000) * 7; // 7Days off-set
+    let dateInMillis = new Date().getUTCMilliseconds() - dateOffset;
+
+    const token = 'someExpiredToken';
+    const mockedDecodedToken = {
+      'sub': 'someSubject',
+      'exp': dateInMillis
+    };
+
+    const stub = sandbox.stub(jwt, 'decode').withArgs(token).callsFake(() => {
+      return mockedDecodedToken;
+    });
+
+    const result = isJWTExpired(token);
+    expect(stub).has.been.calledOnce;
+    expect(result).eq(true);
+  });
+
+  it('checks that a token is valid taking into account the 20 seconds offset', () => {
```

Comment:
The description for the test mentions token is valid but the expectation says returns isJWTExpired = true ?

Reply:
Same as Above :)